ocs_ci/ocs/amq.py

```diff
@@ -0,0 +1,216 @@
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initializer function
+
+        Args:
+            kwargs (dict):
+                Following kwargs are valid
+                namespace: namespace for the operator
+                repo: AMQ repo where all necessary yaml file are there - a github link
+                branch: branch to use from the repo
+        """
+        self.args = kwargs
+        self.repo = self.args.get('repo', constants.OCS_WORKLOADS)
+        self.branch = self.args.get('branch', 'master')
+        self.namespace = self.args.get('namespace', 'my-project')
+        self.amq_is_setup = False
+        self.ocp = OCP()
+        self.ns_obj = OCP(kind='namespace')
+        self.pod_obj = OCP(kind='pod')
+        self.kafka_obj = OCP(kind='Kafka')
+        self.kafka_connect_obj = OCP(kind="KafkaConnect")
+        self.kafka_bridge_obj = OCP(kind="KafkaBridge")
+        self._create_namespace()
+        self._clone_amq()
+
+    def _create_namespace(self):
+        """
+        create namespace for amq
+        """
+        self.ocp.new_project(self.namespace)
+
+    def _clone_amq(self):
+        """
+        clone the amq repo
+        """
+        self.dir = tempfile.mkdtemp(prefix='amq_')
+        try:
+            log.info(f'cloning amq in {self.dir}')
+            git_clone_cmd = f'git clone -b {self.branch} {self.repo} '
+            run(
+                git_clone_cmd,
+                shell=True,
+                cwd=self.dir,
+                check=True
+            )
+            self.amq_dir = "ocs-workloads/amq/v1/install/cluster-operator"
+            self.amq_dir_examples = "ocs-workloads/amq/v1/examples/templates/cluster-operator"
+            self.amq_kafka_pers_yaml = "ocs-workloads/amq/v1/kafka-persistent.yaml"
+            self.amq_kafka_connect_yaml = "ocs-workloads/amq/v1/kafka-connect.yaml"
+            self.amq_kafka_bridge_yaml = "ocs-workloads/amq/v1/kafka-bridge.yaml"
+
+        except (CommandFailed, CalledProcessError)as cf:
+            log.error('Error during cloning of amq repository')
+            raise cf
+
+    def setup_amq_cluster_operator(self):
+        """
+        Function to setup amq-cluster_operator,
+        the file file is pulling from github
+        it will make sure cluster-operator pod is running
+        """
+
+        # self.amq_dir = constants.TEMPLATE_DEPLOYMENT_AMQ_CP
+        run(f'oc apply -f {self.amq_dir} -n {self.namespace}', shell=True, check=True, cwd=self.dir)
+        time.sleep(5)
+        # Wait for strimzi-cluster-operator pod to be created
+        if self.is_amq_pod_running(pod_pattern="cluster-operator"):
+            log.info("strimzi-cluster-operator pod is in running state")
+        else:
+            raise ResourceWrongStatusException("strimzi-cluster-operator pod is not getting to running state")
+
+        run(f'oc apply -f {self.amq_dir_examples} -n {self.namespace}', shell=True, check=True, cwd=self.dir)
```

Comment:
Use exec_oc_cmd() in ocs-ci/ocs_ci/ocs/ocp.py OCP class for executing all oc commands.